### PR TITLE
LPS-127959 No needed to replace empty locale value with the. default value before saving since we do that on front end

### PR DESCRIPTION
**Pull request by marcosapmf**
Related issue: https://issues.liferay.com/browse/LPS-127959

### CODE REVIEW

File: modules/apps/data-engine/data-engine-rest-impl/src/main/java/com/liferay/data/engine/rest/internal/renderer/v2_0/DataLayoutRendererImpl.java [hunk 1]
```diff
@@ -91,6 +91,9 @@ private DDMFormRenderingContext _toDDMFormRenderingContext(
 				dataLayoutRendererContext.getDefaultLanguageId());
 		}
 
+		ddmFormRenderingContext.addProperty(
+			"persisted", dataLayoutRendererContext.isPersisted());
+
```
Comment (jeyvison) [inline modules/apps/data-engine/data-engine-rest-impl/src/main/java/com/liferay/data/engine/rest/internal/renderer/v2_0/DataLayoutRendererImpl.java:96]: remove line

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/java/com/liferay/dynamic/data/mapping/form/renderer/internal/DDMFormFieldTemplateContextFactory.java [hunk 2]
```diff
@@ -753,6 +754,39 @@ protected void setDDMFormFieldTemplateContextValueLocalizableValue(
 		ddmFormFieldTemplateContext.put("localizedValue", localizedValues);
 	}
 
+	protected void setDDMFormFieldTemplateContextValueLocalizableValueEdited(
+		Map<String, Object> ddmFormFieldTemplateContext,
+		DDMFormFieldValue ddmFormFieldValue) {
+
+		Boolean persisted = (Boolean)_ddmFormRenderingContext.getProperty(
```
Comment (jeyvison) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/java/com/liferay/dynamic/data/mapping/form/renderer/internal/DDMFormFieldTemplateContextFactory.java:761]: use getterUtil

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/java/com/liferay/dynamic/data/mapping/form/renderer/internal/DDMFormFieldTemplateContextFactory.java [hunk 3]
```diff
@@ -753,6 +754,39 @@ protected void setDDMFormFieldTemplateContextValueLocalizableValue(
...
+			"persisted");
+
+		if ((persisted == null) || !persisted) {
+			return;
+		}
+
+		if (ddmFormFieldValue == null) {
```
Comment (jeyvison) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/java/com/liferay/dynamic/data/mapping/form/renderer/internal/DDMFormFieldTemplateContextFactory.java:768]: Merge conditions

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/java/com/liferay/dynamic/data/mapping/form/renderer/internal/DDMFormFieldTemplateContextFactory.java [hunk 4]
```diff
@@ -753,6 +754,39 @@ protected void setDDMFormFieldTemplateContextValueLocalizableValue(
...
+			return;
+		}
+
+		Value value = ddmFormFieldValue.getValue();
+
+		if (!(value instanceof LocalizedValue)) {
+			return;
+		}
+
+		Map<String, Object> localizedValueEdited = new HashMap<>();
```
Comment (jeyvison) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/java/com/liferay/dynamic/data/mapping/form/renderer/internal/DDMFormFieldTemplateContextFactory.java:778]: rename to localizedValue

Comment (jeyvison) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/java/com/liferay/dynamic/data/mapping/form/renderer/internal/DDMFormFieldTemplateContextFactory.java:778]: Move to after avaliableLocales, as used

**Comment by marcosapmf** [comment]
ci:test:sf

**Comment by marcosapmf** [comment]
ci:test:relevant

**Comment by marcosapmf** [comment]
ci:test:sf

**Comment by marcosapmf** [comment]
ci:test:relevant

**Comment by marcosapmf** [comment]
ci:test:relevant

**Comment by marcosapmf** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by marcelabc** [comment]
resend: https://github.com/liferay-data-engine/liferay-portal/pull/283

**Pull request closed without merging**